backtrace-library/src/main/java/backtraceio/library/metrics/Event.java

```diff
@@ -0,0 +1,52 @@
+package backtraceio.library.metrics;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import backtraceio.library.common.BacktraceStringHelper;
+
+public abstract class Event {
+
+    @SerializedName("timestamp")
+    protected long timestamp;
+
+    @SerializedName("attributes")
+    protected Map<String, Object> attributes;
+
+    public Event(long timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    public long getTimestamp() {
+        return this.timestamp;
+    }
+
+    public Map<String, Object> getAttributes() {
+        return this.attributes;
```

Comment:
user can store nullable attributes that unique/summed events service will refuse to accept. In this situation, we will lose data. do we filter nullable attributes somewhere?

Reply:
We do filter nullable attributes, but looking back I see we don't call filter method `addAttributesImpl` in UniqueEvents and SummedEvents constructors. I will make this change so we will always filter nullable attributes for UniqueEvents and SummedEvents.